core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,346 @@
+            },
+            F64(rhs) => match Decimal::from_f64_retain(rhs) {
+                Some(x) => match lhs.checked_rem(x) {
+                    Some(y) => Ok(Decimal(y)),
+                    None => Err(Error::OverflowError("%".to_string())),
+                },
+                _ => Err(Error::F64ToDecimalConversionError(rhs.to_string())),
+            },
+            Decimal(rhs) => match lhs.checked_rem(rhs) {
+                Some(x) => Ok(Decimal(x)),
+                _ => Err(Error::F64ToDecimalConversionError(rhs.to_string())),
+            },
+
+            Null => Ok(Null),
+            _ => Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(lhs),
+                operator: NumericBinaryOperator::Modulo,
+                rhs: rhs.clone(),
+            }
+            .into()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::{TryBinaryOperator, Value::*},
+        crate::data::{NumericBinaryOperator, ValueError},
+        rust_decimal::prelude::Decimal,
+        std::cmp::Ordering,
+    };
+
+    #[test]
+    fn eq() {
+        let base = Decimal::ONE;
+
+        assert_eq!(base, I8(1));
+        assert_eq!(base, I64(1));
+        assert_eq!(base, F64(1.0));
+        assert_eq!(base, Decimal(Decimal::ONE));
+
+        assert_ne!(base, Bool(true));
+    }
+
+    #[test]
+    fn partial_cmp() {
+        let base = Decimal::ONE;
+
+        assert_eq!(base.partial_cmp(&I8(1)), Some(Ordering::Equal));
+        assert_eq!(base.partial_cmp(&I64(1)), Some(Ordering::Equal));
+        assert_eq!(base.partial_cmp(&F64(1.0)), Some(Ordering::Equal));
+        assert_eq!(
+            base.partial_cmp(&Decimal(Decimal::ONE)),
+            Some(Ordering::Equal)
+        );
+
+        assert_eq!(base.partial_cmp(&Bool(true)), None);
+    }
+
+    #[test]
+    fn try_add() {
+        let dec_epislon: Decimal = Decimal::from_f64_retain(f64::EPSILON).unwrap();
+        let base = Decimal::ONE;
+
+        assert!(
+            matches!(base.try_add(&I8(1)), Ok(Decimal(x)) if (x - Decimal::TWO).abs() < dec_epislon )
+        );
+        assert!(
+            matches!(base.try_add(&I64(1)), Ok(Decimal(x)) if (x - Decimal::TWO).abs() < dec_epislon )
+        );
+        assert!(
+            matches!(base.try_add(&F64(1.0)), Ok(Decimal(x)) if (x - Decimal::TWO).abs() < dec_epislon )
+        );
+        assert!(
+            matches!(base.try_add(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if (x - Decimal::TWO).abs() < dec_epislon )
+        );
+
+        assert_eq!(
+            base.try_add(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(base),
+                operator: NumericBinaryOperator::Add,
+                rhs: Bool(true).clone(),
+            }
+            .into()),
+        );
+    }
+
+    #[test]
+    fn try_subtract() {
+        let dec_epislon: Decimal = Decimal::from_f64_retain(f64::EPSILON).unwrap();
```

Comment:
In a Decimal type, wouldn't it be okay not to compare with epsilon values?

Reply:
Why was the F64 comparison done using epsilon?
Answer) <https://rust-lang.github.io/rust-clippy/master/#float_cmp>